### PR TITLE
WASM: Implement binary instructions class

### PR DESCRIPTION
**Pull request by Shaikh-Ubaid**
towards https://github.com/lfortran/lfortran/issues/589, https://github.com/lcompilers/lpython/issues/1485

### CODE REVIEW

**Comment by Shaikh-Ubaid** [comment]
This is ready. Please review and share feedback.

**Review by czgdp1807 — APPROVED** [review]

File: src/libasr/codegen/wasm_assembler.h [hunk 1]
```diff
@@ -904,345 +243,6 @@ void save_bin(Vec<uint8_t> &code, std::string filename) {
...
-    emit_u32(code, al, mem_offset);
-}
-
-// function to emit i64.store32 instruction
-void emit_i64_store32(Vec<uint8_t> &code, Allocator &al, uint32_t mem_align,
-                      uint32_t mem_offset) {
-    emit_b8(code, al, 0x3E);
-    emit_u32(code, al, mem_align);
-    emit_u32(code, al, mem_offset);
-}
```
Comment (certik) [inline src/libasr/codegen/wasm_assembler.h:1244]: All these are now generated automatically from `wasm_intructions_visitor.py`?

**Review by certik — APPROVED** [review]
I think this looks good overall, especially if it removes so much code that is now automatically generated.

File: src/libasr/codegen/wasm_assembler.h (hunk 1 above)
Comment (Shaikh-Ubaid) [reply src/libasr/codegen/wasm_assembler.h:1244]: Yes, they are automatically generated.

Comment (certik) [reply src/libasr/codegen/wasm_assembler.h:1244]: That's excellent. Much less code to maintain.

**Pull request merged**